lib/alma/loan_set.rb

```diff
@@ -4,12 +4,52 @@ module Alma
   class LoanSet < ResultSet
     alias :total_records :total_record_count
 
+
+    attr_reader :results, :raw_response
+    def_delegators :results, :empty?
+
+    def initialize(raw_response)
+      @raw_response = raw_response
+      @response = JSON.parse(raw_response.body)
+      @results = @response.fetch(key, [])
+        .map { |item| single_record_class.new(item) }
+    end
+
+
+    def all
+      Enumerator.new do |yielder|
+        offset = 0
+        loop do
+          r = (offset == 0) ? self : single_record_class.where_user(user_id, {limit: 100, offset: offset})
+          unless r.empty?
+            r.map { |item| yielder << item }
+            offset += 100
+          else
+            raise StopIteration
```

Comment:
I think that this pattern requires you to throw and error not raise an exception as raising exceptions is so supposed to be about actual errors not flow control.  That said, I'm not sure if this is a requirement or just convention.

Reply:
"An Enumerator can also be used as an external iterator. For example, #next returns the next value of the iterator or raises StopIteration if the Enumerator is at the end." from https://ruby-doc.org/core-2.2.0/Enumerator.html

Reply:
I was going to post; nevermind I just read up on Enumerators at https://ruby-doc.org/core-2.2.0/Enumerator.html and they specifically use raise not throw.  I'm not sure where I picked up that throw vs raise lore.